feat(ui): enhance empty state handling for executions and selection in components

sikehish opened the pull request:
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->

### What changes are being made and why?

This PR introduces two main features to improve user experience when interacting with data tables and charts in the application:

1. **Empty State Handling**: Added an `<el-empty>` component to both the Bar and SelectTable components. This component is displayed when there are zero executions or no data available, enhancing the user interface by providing clear feedback to users.

2. **Conditional Rendering**: Updated the rendering logic for the bulk select header in the SelectTable component to only show when there is data selected, ensuring a cleaner interface and preventing confusion when no data is available.

These changes enhance usability by clearly communicating to users when there is no data to display, thereby improving the overall experience.

---

### How the changes have been QAed?

To QA the changes, the following flow demonstrates the user experience when there are zero executions or no data:

1. Navigate to Executions, or Flows.
2. Ensure that there are no executions(or flows). 
3. Observe that the `<el-empty>` component is correctly displayed in both the Bar and SelectTable components.

```yaml
- Navigate to Executions, or Flows.
- If any executions/flows are present, you may delete them.
- Confirm that the empty state message is displayed correctly.
```

sikehish committed:
feat(select-table): add conditional rendering for empty state and selection header

- Added an <el-empty> component to display when data is not available
- Updated bulk select header to render only if there is selection and data

sikehish committed:
feat(bar): add empty state for zero executions

- Added <el-empty> component to display when total executions are 0
- Updated conditional rendering for info container and chart display based on total executions

MilosPaunovic commented on ui/src/components/layout/SelectTable.vue:
```diff
@@ -1,16 +1,27 @@
     </div>
 </template>
 
+
```
Remove this extra line.

MilosPaunovic reviewed (changes requested):
This is great work so far @sikehish.

I have one small cleanup comment on this so far, but i'd like to ask you to add the same logic you've added to `executions/Bar.vue` component to the `executions/Doughnut.vue` and to `logs/Bar.vue`, to cover all of the new charts.

Also, in `kestra/ui/src/components/executions/Executions.vue` I would prefer to add `<template #table v-if="executions.length">` this to line 143, and in `kestra/ui/src/components/flows/Flows.vue` something similar, `<template #table v-if="flows.length">` to line 89, as it would look nicer if we don't ave two empty states on the same page, one below the other.

Skraye commented:
Also, can you share a screen of what it looks like ? 
Is this displayed? 
![image](https://github.com/user-attachments/assets/a45cc50e-2a2d-4e3e-8370-791542dfc618)

If yes, it needs to use the translation

sikehish commented:
> This is great work so far @sikehish.
> 
> I have one small cleanup comment on this so far, but i'd like to ask you to add the same logic you've added to `executions/Bar.vue` component to the `executions/Doughnut.vue` and to `logs/Bar.vue`, to cover all of the new charts.
> 
> Also, in `kestra/ui/src/components/executions/Executions.vue` I would prefer to add `<template #table v-if="executions.length">` this to line 143, and in `kestra/ui/src/components/flows/Flows.vue` something similar, `<template #table v-if="flows.length">` to line 89, as it would look nicer if we don't ave two empty states on the same page, one below the other.

Thank you, will get this done ASAP :)

sikehish commented:
> Also, can you share a screen of what it looks like ? Is this displayed? ![image](https://private-user-images.githubusercontent.com/37600690/378407832-a45cc50e-2a2d-4e3e-8370-791542dfc618.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MTE4MTMsIm5iZiI6MTcyOTUxMTUxMywicGF0aCI6Ii8zNzYwMDY5MC8zNzg0MDc4MzItYTQ1Y2M1MGUtMmEyZC00ZTNlLTgzNzAtNzkxNTQyZGZjNjE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDExNTE1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZiY2ZmMjBhYWY0NWJkNWFmYzhmOTcyNjI2MDFkZWM5ZmRkNDJlNDRkYjUzZmJjN2FhYzgxZDYxMjU1NTQ4ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mat0PSRbNe9jLxiNy-e5bU5k3M0XiH9aDXPLFn9OIQw)
> 
> If yes, it needs to use the translation

Could you guide me with the translation part? I'll share the image in a while.

Skraye commented:
Sure, its something straightforward, first here is how you use it, with the `$t()` methods :
https://github.com/kestra-io/kestra/blob/51b78a963acb537f6a1f5d85d2509596deb912a7/ui/src/components/flows/FlowConcurrency.vue#L39-L39

And all you have to do, is to put the english translation here :
https://github.com/kestra-io/kestra/blob/51b78a963acb537f6a1f5d85d2509596deb912a7/ui/src/translations/en.json#L893-L893

Our CI will handle translation in every other language :+1:

sikehish committed:
Merge branch 'kestra-io:develop' into develop

sikehish committed:
feat(ui): improve empty state display for executions and flows tables

- Added conditional rendering for the executions table in Executions.vue to avoid displaying two empty states on the same page.
- Implemented similar changes for the flows table in Flows.vue.

sikehish committed:
feat(doughnut): add total prop to Doughnut.vue and implement execution logic

- Added a new `total` prop to the Doughnut.vue component with type Number.
- Implemented similar logic from executions/Bar.vue into executions/Doughnut.vue for consistency in handling total executions.

sikehish committed:
feat(logs-wrapper): add el-empty component and enhance validation

- Added an `<el-empty>` component to display when `hasStatsData` is false.
- Enhanced validation logic in `<template #table>` on line 58 for improved data handling.

sikehish committed:
feat(translations): add English translation for "no data"

- Added translation for "no data" corresponding to "No data available" in `kestra/ui/src/translations/en.json`.

sikehish commented:
> Sure, its something straightforward, first here is how you use it, with the `$t()` methods :
> 
> https://github.com/kestra-io/kestra/blob/51b78a963acb537f6a1f5d85d2509596deb912a7/ui/src/components/flows/FlowConcurrency.vue#L39-L39
> 
> And all you have to do, is to put the english translation here :
> 
> https://github.com/kestra-io/kestra/blob/51b78a963acb537f6a1f5d85d2509596deb912a7/ui/src/translations/en.json#L893-L893
> 
> Our CI will handle translation in every other language 👍

Hi @Skraye. I've added the translation. Let me know if anything else needs to be added/modified.

sikehish committed:
feat(logs): add conditional rendering for logs in Bar.vue

- Wrapped the existing Bar component in a div that checks if `props.data.length > 0`.
- Added an <el-empty> component.

sikehish commented:
> This is great work so far @sikehish.
> 
> I have one small cleanup comment on this so far, but i'd like to ask you to add the same logic you've added to `executions/Bar.vue` component to the `executions/Doughnut.vue` and to `logs/Bar.vue`, to cover all of the new charts.
> 
> Also, in `kestra/ui/src/components/executions/Executions.vue` I would prefer to add `<template #table v-if="executions.length">` this to line 143, and in `kestra/ui/src/components/flows/Flows.vue` something similar, `<template #table v-if="flows.length">` to line 89, as it would look nicer if we don't ave two empty states on the same page, one below the other.

Hi @MilosPaunovic 
Ive implemented all of this.
However, I've made some changes in LogsWrapper.vue , as the logic and validation was already in place at the LogsWrapper level. I've also made changes  in the logs/Bar.vue component( basically added an additional check, as it's being rendered in Dashboard). I realised that  Bar.vue was being conditionally rendered in LogsWrapper so I modified the else condition in LogsWrapper.

MilosPaunovic committed:
Update en.json

MilosPaunovic committed:
Update LogsWrapper.vue

MilosPaunovic committed:
Update SelectTable.vue

MilosPaunovic committed:
Update Bar.vue

MilosPaunovic committed:
Update Doughnut.vue

MilosPaunovic committed:
Update Bar.vue

MilosPaunovic commented:
Just a note here, you've done a great job, I've just pushed small tweaks to have everything properly aligned.
